### PR TITLE
prser: add alter user default role support

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
support gramma like `ALTER USER 'root'@'localhost' DEFAULT ROLE r1`
reference: https://dev.mysql.com/doc/refman/8.0/en/alter-user.html#alter-user-role

### What is changed and how it works?
`ALTER USER ... DEFAULT ROLE` is much like `SET DEFAULT ROLE`.
So they use the same structure `SetDefaultRoleStmt`.
Adding `IsAlter` and `IfExist` to support `Restore` and `IF EXISTS` option.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -6098,6 +6098,19 @@ SetDefaultRoleStmt:
 			UserList: $6.([]*auth.UserIdentity),
 		}
 	}
+| 	"ALTER" "USER" IfExists Username "DEFAULT" "ROLE" SetDefaultRoleOpt
+	{
+		user := make([]*auth.UserIdentity, 0, 1)
+		user = append(user, $4.(*auth.UserIdentity))
```
Comment (kennytm) [inline parser.y]:
```go
user := []*auth.UserIdentity{$4.(*auth.UserIdentity)}
```

File: parser.y [hunk 2]
```diff
@@ -6098,6 +6098,19 @@ SetDefaultRoleStmt:
...
+		tmp := $7.(*ast.SetRoleStmt)
+		$$ = &ast.SetDefaultRoleStmt{
+        	SetRoleOpt: tmp.SetRoleOpt,
+        	RoleList: tmp.RoleList,
+        	IsAlter: true,
+        	IfExist: $3.(bool),
+        	UserList: user,
+        }
```
Comment (kennytm) [inline parser.y]: Fix indentation of lines 6107–6112 (reindent with tabs).

Comment (imtbkcat) [reply parser.y]: ok

**Comment by zz-jason** [comment]
@imtbkcat any update?

**Comment by imtbkcat** [comment]
@zz-jason PTAL

**Pull request closed without merging**